SwiftDungeon/EnemyFactory.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  EnemyFactory.swift
+//  SwiftDungeon
+//
+//  Created by Toro Juan D. on 3/20/18.
+//  Copyright © 2018 Toro Juan D. All rights reserved.
+//
+
+import Foundation
+
+class EnemyFactory {
+    
+    func createEnemy() -> Enemy {
+        return Enemy()
+    }
+    
```

Comment:
Shouldn't need a factory in this case. Can always just make an Enemy. Factories are useful when you have many variations of something and many inputs for them. The factory should handle giving you back the right one based on the inputs. Think of it like raw material in -> finished product out. Where the raw materials are the dependencies and the finished product is the instance.

Reply:
In the above example you always get back an enemy but they can be different instances of an enemy initialized with different types and different Ai's to behave differently.